backend/src/game/game.gateway.ts

```diff
@@ -0,0 +1,252 @@
+      ball.vel.y *= -1;
+    } else if (ball.pos.y >= gameWinHght - ballPx && ball.vel.y > 0) {
+      ball.vel.y *= -1;
+    } else if (
+      // goal hit
+      ball.pos.x <= 0 ||
+      ball.pos.x >= gameWinWid - ballPx
+    ) {
+      if (ball.vel.x < 0) {
+        rightPaddle.score++;
+      } else {
+        leftPaddle.score++;
+      }
+      ball.pos = deepCpInitBall().pos;
+      ball.vel = deepCpInitBall().vel;
+      ball.vel.x *= -1;
+    } else if (
+      // left paddle hit
+      ball.vel.x < 0 &&
+      this.isHitPaddle(ball, leftPaddle)
+    ) {
+      this.handlePaddleCollision(ball, leftPaddle);
+    } else if (
+      // right paddle hit
+      ball.vel.x > 0 &&
+      this.isHitPaddle(ball, rightPaddle)
+    ) {
+      this.handlePaddleCollision(ball, rightPaddle);
+    }
+    return ball;
+  }
+
+  // ballの処理↑
+
+  @WebSocketServer()
+  private server: Server;
+  private intervalId: NodeJS.Timeout;
+  private logger: Logger = new Logger('GameGateway');
+
+  afterInit(_server: Server) {
+    //初期化
+    this.logger.log('初期化しました。');
+  }
+
+  private loop() {
+    let lastFrameTime = Date.now();
+    this.intervalId = setInterval(() => {
+      const currentTime = Date.now();
+      const deltaTime = (currentTime - lastFrameTime) / 1000;
+      lastFrameTime = currentTime;
+      serverBall = this.updateBall(
+        serverBall,
+        deltaTime,
+        400,
+        playersPaddle.get(leftPlayerId),
+        playersPaddle.get(rightPlayerId),
+      );
+
+      // クライアントにデータを送信する
+      this.server.emit('updateBall', serverBall);
+      this.server.emit('updatePaddle', Object.fromEntries(playersPaddle));
+    }, 1000 / 60);
+  }
+
+  handleConnection(@ConnectedSocket() client: Socket) {
+    //クライアント接続時
+    this.logger.log(`Client connected: ${client.id}`);
+    if (playersPaddle.size === 0) {
+      this.logger.log('first player joined');
+      //creating player 1
+      playersPaddle.set(client.id, initLeftPaddle);
+      playersProfile.set(client.id, initLeftProfile);
+      leftPlayerId = client.id;
+    } else if (playersPaddle.size === 1) {
+      this.logger.log('second player joined');
+      //creating player 2
+      playersPaddle.set(client.id, initRightPaddle);
+      playersProfile.set(client.id, initRightProfile);
+      rightPlayerId = client.id;
+      this.server.emit('updatePaddle', Object.fromEntries(playersPaddle));
+      serverBall = deepCpInitBall();
+      this.loop();
+    } else if (playersPaddle.size > 1) {
+      this.logger.log('too many players, bye');
+    }
+  }
+
+  handleDisconnect(@ConnectedSocket() client: Socket) {
+    this.logger.log(`Client disconnected: ${client.id}`);
+    clearInterval(this.intervalId);
+  }
+
+  @SubscribeMessage('updatePaddle')
+  handleUpdatePaddle(@MessageBody() newPaddle: IPaddle): void {
+    if (newPaddle.id === 'left') playersPaddle.set(leftPlayerId, newPaddle);
+    else playersPaddle.set(rightPlayerId, newPaddle);
+  }
+  //updatereadybutton
```

Comment:
細かいですが、コメントなので可読性のためにスペースを開けたいです。

```suggestion
  //update ready button
```

Reply:
対応完了